### PR TITLE
APPS-1092 Enhance backup metadata

### PR DESCRIPTION
**Pull request by korotkov-aerospike**

### CODE REVIEW

**Comment by korotkov-aerospike** [comment]
http://3.65.168.247:8080/ui/#pitr/launches/all/4946
tests passed

**Comment by korotkov-aerospike** [comment]
Screenshot 2024-01-22 at 14 41 02
that's how the result look

File: pkg/service/backup_backend_s3.go [hunk 1]
```diff
@@ -56,24 +56,23 @@ func (s *BackupBackendS3) writeState(state *model.BackupState) error {
...
 		if s.fullBackupInProgress.Load() {
 			return []model.BackupDetails{}, nil
 		}
-		if lastRun.UnixMilli() < from || lastRun.UnixMilli() >= to {
+		metadata, err := s.readMetadata(backupFolder)
+		if err != nil {
+			return []model.BackupDetails{}, err
+		}
+		if metadata.Created.UnixMilli() < from ||
+			metadata.Created.UnixMilli() > to {
```
Comment (reugn) [inline pkg/service/backup_backend_s3.go:70]: `>= to` inclusive?

File: pkg/service/backup_backend_s3.go [hunk 2]
```diff
@@ -56,24 +56,23 @@ func (s *BackupBackendS3) writeState(state *model.BackupState) error {
 func (s *BackupBackendS3) FullBackupList(from, to int64) ([]model.BackupDetails, error) {
 	backupFolder := s.Path + "/" + model.FullBackupDirectory + "/"
 	s3prefix := "s3://" + s.bucket
-	lastRun := s.readState().LastFullRun
-	slog.Info("get full backups", "backupFolder", backupFolder, "lastRun", lastRun, "from", from, "to", to)
+	slog.Info("get full backups", "backupFolder", backupFolder, "from", from, "to", to)
```
Comment (reugn) [inline pkg/service/backup_backend_s3.go:59]: Capitalize get

File: pkg/service/backup_backend_local.go [hunk 3]
```diff
@@ -182,45 +192,19 @@ func (local *BackupBackendLocal) DeleteFolder(path string) error {
...
-	if err != nil {
-		slog.Error("failed to calculate size", "path", path, "err", err)
-		return nil
-	}
-
-	return &size
-}
-
 func (local *BackupBackendLocal) writeBackupMetadata(path string, metadata model.BackupMetadata) error {
+	slog.Info("Write backup metadata", "data", metadata)
```
Comment (reugn) [inline pkg/service/backup_backend_local.go:207]: Can be removed?

File: pkg/service/backup_backend_local.go [hunk 4]
```diff
@@ -96,49 +95,59 @@ func (local *BackupBackendLocal) writeState(state *model.BackupState) error {
...
 			return []model.BackupDetails{}, nil
 		}
-		// check request time boundaries
-		if lastRun.UnixMilli() < from || lastRun.UnixMilli() >= to {
+		details, err := local.toBackupDetails(backupFolder)
+		if err != nil {
+			return nil, err
+		}
+		if details.Created.UnixMilli() < from ||
+			details.Created.UnixMilli() > to {
```
Comment (reugn) [inline pkg/service/backup_backend_local.go:115]: `>=`?

**Review by reugn — APPROVED** [review]

**Pull request merged**